test/helpers/utils.go

```diff
@@ -1199,3 +1199,60 @@ func WriteFileToNode(t *testing.T, cs *framework.ClientSet, node corev1.Node, fi
 		ExecCmdOnNode(t, cs, node, "rm", filename)
 	})
 }
+
+// Polls the ControllerConfig and calls the provided condition function with
+// the ControllerConfig to determine if the ControllerConfig has reached the
+// expected state.
+func AssertControllerConfigReachesExpectedState(t *testing.T, cs *framework.ClientSet, condFunc func(*mcfgv1.ControllerConfig) bool) {
+	t.Helper()
+
+	start := time.Now()
+	err := wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, 5*time.Minute, true, func(pCtx context.Context) (bool, error) {
+		cc, err := cs.ControllerConfigs().Get(pCtx, "machine-config-controller", metav1.GetOptions{})
+		require.NoError(t, err)
+
+		return condFunc(cc), nil
+	})
+
+	require.NoError(t, err, "ControllerConfig failed to reach expected state")
+	t.Logf("ControllerConfig reached expected state in %v", time.Since(start))
+}
+
+// Polls all of the nodes and calls the provided condition function with each
+// node to determine if the ControllerConfig has reached the expected state.
```

Comment:
This function can be used to with any sort of condition check local to this node - pretty neat! So I don't think calling out ControllerConfig here is necessary, as it is specifically related to the test we are adding here